### DISCUSSION TITLE
Questions about using allegro's kokkos acceleration

### DISCUSSION THREAD

**Discussion by chenmao001** [mir-group/allegro#15] [Q&A] [answered]
Hello, dear developers，After I tried some operations, I encountered some problems:

I used **LAMMPS+allegro** and **LAMMPS+kokkos+allegro** to simulate the atoms respectively.  From the log of LAMMPS, I can see that using kokkos did not accelerate the effect, but increased the comm time in lammps. This is what reason?
Is my compile option wrong?  I refer to: https://github.com/mir-group/pair_allegro
Have you ever encountered the above situation? How should I solve this problem?

**Comment by Linux-cpp-lisp** [comment, marked as answer]
Hi @chenmao001 ,

Kokkos is only helpful if you are using a GPU.  If you are using a GPU, this is likely due to a system-specific issue with how Kokkos and communication libraries were compiled on your platform... this can be quite finicky and sometimes using things that should be faster like GPU-GPU MPI can fail to help if setup incorrectly on the system or compilation.

I would recommend doing some tests on other Kokkos pair styles built into LAMMPS and consulting your system administrator.
